io.audio.MasVisGtk.json

```diff
@@ -0,0 +1,389 @@
+        "*.la",
+        "*.a"
+    ],
+    "modules" : [
+		{
+			"name": "python3-meson-python",
+			"buildsystem": "simple",
+			"build-commands": [
+				"pip3 install --verbose --exists-action=i --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} \"meson-python\" --no-build-isolation"
+			],
+			"sources": [
+				{
+				    "type": "file",
+				    "url": "https://files.pythonhosted.org/packages/91/c0/104cb6244c83fe6bc3886f144cc433db0c0c78efac5dc00e409a5a08c87d/meson_python-0.16.0-py3-none-any.whl",
+				    "sha256": "842dc9f5dc29e55fc769ff1b6fe328412fe6c870220fc321060a1d2d395e69e8"
+				},
+				{
+				    "type": "file",
+				    "url": "https://files.pythonhosted.org/packages/08/aa/cc0199a5f0ad350994d660967a8efb233fe0416e4639146c089643407ce6/packaging-24.1-py3-none-any.whl",
+				    "sha256": "5b8f2217dbdbd2f7f384c41c628544e6d52f2d0f53c6d0c3ea61aa5d1d7ff124"
+				},
+				{
+				    "type": "file",
+				    "url": "https://files.pythonhosted.org/packages/aa/5f/bb5970d3d04173b46c9037109f7f05fc8904ff5be073ee49bb6ff00301bc/pyproject_metadata-0.8.0-py3-none-any.whl",
+				    "sha256": "ad858d448e1d3a1fb408ac5bac9ea7743e7a8bbb472f2693aaa334d2db42f526"
+				}
+			]
+		},
+		{
+			"name": "python3-typing-extensions",
+			"buildsystem": "simple",
+			"build-commands": [
+				"pip3 install --verbose --exists-action=i --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} \"typing_extensions\" --no-build-isolation"
+			],
+			"sources": [
+				{
+				    "type": "file",
+				    "url": "https://files.pythonhosted.org/packages/26/9f/ad63fc0248c5379346306f8668cda6e2e2e9c95e01216d2b8ffd9ff037d0/typing_extensions-4.12.2-py3-none-any.whl",
+				    "sha256": "04e5ca0351e0f3f85c6853954072df659d0d13fac324d0072316b67d7794700d"
+				}
+			]
+		},
+		{
+			"name": "python3-setuptools-scm",
+			"buildsystem": "simple",
+			"build-commands": [
+				"pip3 install --verbose --exists-action=i --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} \"setuptools-scm\" --no-build-isolation"
+			],
+			"sources": [
+				{
+				    "type": "file",
+				    "url": "https://files.pythonhosted.org/packages/a0/b9/1906bfeb30f2fc13bb39bf7ddb8749784c05faadbd18a21cf141ba37bff2/setuptools_scm-8.1.0-py3-none-any.whl",
+				    "sha256": "897a3226a6fd4a6eb2f068745e49733261a21f70b1bb28fce0339feb978d9af3"
+				}
+			]
+		},
+		{
+			"name": "python3-pybind11",
+			"buildsystem": "simple",
+			"build-commands": [
+				"pip3 install --verbose --exists-action=i --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} \"pybind11\" --no-build-isolation"
+			],
+			"sources": [
+				{
+				    "type": "file",
+				    "url": "https://files.pythonhosted.org/packages/06/55/9f73c32dda93fa4f539fafa268f9504e83c489f460c380371d94296126cd/pybind11-2.11.1-py3-none-any.whl",
+				    "sha256": "33cdd02a6453380dd71cc70357ce388ad1ee8d32bd0e38fc22b273d050aa29b3"
+				}
+			]
+		},
+		{
+			"name": "python3-openblas-lapack",
```

Comment:
This should be in the main manifest

Reply:
This may be a problem. `numpy`, (and maybe) `scipy` and `matplotlib` need to have this with them, for optimal computation.

Reply:
I didn't tell you remove it completely, install these in the main manifest, separate all the python parts and place it _after_ these dependencies.

Reply:
`python3-openblas-lapack` is a computational package, why would you put that in the main manifest, but `numpy` and the others in `python3-deps.json`?

Reply:
Because neither of blas or lapack are being generated by flatpak-pip-generator. These are system dependencies.

Reply:
As you know, if the order is wrong, the package may not compile at all. So doees that mean 16 json files?
Then to make it simple, which of the following packages goes to a separate file `python3-deps` ?

```
python3-meson-python
python3-typing-extensions
python3-setuptools-scm
python3-pybind11
python3-openblas-lapack
python3-numpy
python3-matplotlib-deps
    python3-contourpy
    python3-flit-core
    python3-tomli
    python3-certifi
    python3-cppy
    python3-Pillow
    python3-cycler
    python3-kiwisolver
    python3-pyparsing
    python3-python_dateutil
python3-pythran
python3-scipy
python3-matplotlib
```

Reply:
No in your main manifest:

```json
				{
				    "name": "openblas",
				    "no-autogen": true,
				    "make-args": [
				        "DYNAMIC_ARCH=1",
				        "FC=gfortran",
				        "NO_CBLAS=1",
				        "NO_LAPACKE=1",
				        "USE_OPENMP=0"
				    ],
				    "make-install-args": [
				        "PREFIX=/app"
				    ],
				    "sources": [
				        {
				            "type": "archive",
				            "url": "https://github.com/OpenMathLib/OpenBLAS/archive/refs/tags/v0.3.28.tar.gz",
				            "sha256": "f1003466ad074e9b0c8d421a204121100b0751c96fc6fcf3d1456bd12f8a00a1"
				        }
				    ]
				},
				{
				    "name": "lapack",
				    "buildsystem": "cmake-ninja",
				    "builddir": true,
				    "cleanup": [
				        "/lib/cmake"
				    ],
				    "config-opts": [
				        "-DCMAKE_BUILD_TYPE=RelWithDebInfo",
				        "-DBUILD_SHARED_LIBS=ON",
				        "-DBUILD_TESTING=OFF",
				        "-DLAPACKE=ON",
				        "-DCBLAS=ON"
				    ],
				    "sources": [
				        {
				            "type": "archive",
				            "url": "https://github.com/Reference-LAPACK/lapack/archive/refs/tags/v3.12.0.tar.gz",
				            "sha256": "eac9570f8e0ad6f30ce4b963f4f033f0f643e7c3912fc9ee6cd99120675ad48b"
				        }
				    ]
				}
			]
		},
    {
			"type": "archive",
			"url": "https://gitlab.freedesktop.org/freetype/freetype/-/archive/VER-2-6-1/freetype-VER-2-6-1.tar.gz",
			"sha256": "85764250282ec0b8ff2918f48a5c462be4d2089456e3e1d65f1bd6648cb5000c"
     },
		{
			"type": "archive",
			"url": "https://github.com/qhull/qhull/archive/refs/tags/2020.2.tar.gz",
				"dest": "build/qhull-2020.2",
				"sha256": "59356b229b768e6e2b09a701448bfa222c37b797a84f87f864f97462d8dbc7c5"
		},
    "python3-deps.json"
```

where python3-deps contains only the python packages you generate with flatpak-pip-generator.

Reply:
What about `qhull` and `freetype`?

Reply:
Add them after lapack

Reply:
How do you build them. There seem to be errors, when moving them as their own module {}. Doesn't `matplotlib` want them inside so that they are "baked into the code"?

```
========================================================================
Building module freetype in /home/<USERNAME>/.var/app/org.gnome.Builder/cache/gnome-builder/flatpak-builder/build/freetype-1
========================================================================
Running: python3 setup.py install --prefix=/app --root=/
python3: can't open file '/run/build/freetype/setup.py': [Errno 2] No such file or directory
Error: module freetype: Child process exited with code 2
```

Do you know how to build the latest `matplotlib=3.9.2`, `numpy`, `scipy` ?

Reply:
Freetype should be in the runtime already, you don't have to build it here. Just pass
`-Dsystem-freetype=true` to matplotlib

See https://github.com/flathub/org.artisan_scope.artisan/blob/9d5f356d7833549d99de9a60446c437587c1bb9e/dep-python3-source.json#L10

Reply:
`freetype` and `qhull` seem to compile with your good advice.

`openblas` is in the main json.

`python3-scipy` from `python3-deps.json` fails 

```
		{
		    "name": "python3-scipy",
		    "buildsystem": "simple",
		    "build-commands": [
		        "pip3 install --verbose --exists-action=i --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} \"scipy\" --no-build-isolation"
		    ],
		    "sources": [
		        {
		            "type": "file",
		            "url": "https://files.pythonhosted.org/packages/30/85/cdbf2c3c460fe5aae812917866392068a88d02f07de0fe31ce738734c477/scipy-1.12.0.tar.gz",
		            "sha256": "4bf5abab8a36d20193c698b0f1fc282c1d083c94723902c447e5d2f1780936a3"
		        }
		    ]
		},
```

Error

```
========================================================================
Building module python3-scipy in /home/<USERNAME>/.var/app/org.gnome.Builder/cache/gnome-builder/flatpak-builder/build/python3-scipy-1
========================================================================
Running: pip3 install --verbose --exists-action=i --no-index --find-links="file://${PWD}" --prefix=${FLATPAK_DEST} "scipy" --no-build-isolation
Using pip 24.1.2 from /usr/lib/python3.11/site-packages/pip (python 3.11)
Looking in links: file:///run/build/python3-scipy
Processing ./scipy-1.12.0.tar.gz
  Running command Preparing metadata (pyproject.toml)
  + meson setup /tmp/pip-install-ytc4zjg6/scipy_1ce7c2a53d454c308c1382a557048dc7 /tmp/pip-install-ytc4zjg6/scipy_1ce7c2a53d454c308c1382a557048dc7/.mesonpy-zjne_yxy -Dbuildtype=release -Db_ndebug=if-release -Db_vscrt=md --native-file=/tmp/pip-install-ytc4zjg6/scipy_1ce7c2a53d454c308c1382a557048dc7/.mesonpy-zjne_yxy/meson-python-native-file.ini
  The Meson build system
  Version: 1.3.2
  Source dir: /tmp/pip-install-ytc4zjg6/scipy_1ce7c2a53d454c308c1382a557048dc7
  Build dir: /tmp/pip-install-ytc4zjg6/scipy_1ce7c2a53d454c308c1382a557048dc7/.mesonpy-zjne_yxy
  Build type: native build
  Project name: scipy
  Project version: 1.12.0
  C compiler for the host machine: ccache cc (gcc 13.2.0 "cc (GCC) 13.2.0")
  C linker for the host machine: cc ld.bfd 2.42
  C++ compiler for the host machine: ccache c++ (gcc 13.2.0 "c++ (GCC) 13.2.0")
  C++ linker for the host machine: c++ ld.bfd 2.42
  Cython compiler for the host machine: cython (cython 0.29.37)
  Host machine cpu family: x86_64
  Host machine cpu: x86_64
  Program python found: YES (/usr/bin/python)
  Found pkg-config: YES (/usr/bin/pkg-config) 2.2.0
  Run-time dependency python found: YES 3.11
  Program cython found: YES (/usr/bin/cython)
  Compiler for C supports arguments -Wno-unused-but-set-variable: YES
  Compiler for C supports arguments -Wno-unused-function: YES
  Compiler for C supports arguments -Wno-conversion: YES
  Compiler for C supports arguments -Wno-misleading-indentation: YES
  Library m found: YES
  Fortran compiler for the host machine: gfortran (gcc 13.2.0 "GNU Fortran (GCC) 13.2.0")
  Fortran linker for the host machine: gfortran ld.bfd 2.42
  Compiler for Fortran supports arguments -Wno-conversion: YES
  Checking if "-Wl,--version-script" : links: YES
  Program pythran found: YES (/app/bin/pythran)
  Found CMake: /usr/bin/cmake (3.29.6)
  WARNING: CMake Toolchain: Failed to determine CMake compilers state
  Run-time dependency xsimd found: NO (tried pkgconfig and cmake)
  Run-time dependency threads found: YES
  Library npymath found: YES
  Library npyrandom found: YES
  pybind11-config found: YES (/app/bin/pybind11-config) 2.11.1
  Run-time dependency pybind11 found: YES 2.11.1
  Run-time dependency scipy-openblas found: NO (tried pkgconfig)
  Run-time dependency openblas found: NO (tried pkgconfig and cmake)
  Run-time dependency openblas found: NO (tried pkgconfig and cmake)

  ../scipy/meson.build:169:9: ERROR: Dependency "OpenBLAS" not found, tried pkgconfig and cmake

  A full log can be found at /tmp/pip-install-ytc4zjg6/scipy_1ce7c2a53d454c308c1382a557048dc7/.mesonpy-zjne_yxy/meson-logs/meson-log.txt
  error: subprocess-exited-with-error
  
  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /usr/bin/python /usr/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py prepare_metadata_for_build_wheel /tmp/tmp2ivqrx0r
  cwd: /tmp/pip-install-ytc4zjg6/scipy_1ce7c2a53d454c308c1382a557048dc7
  Preparing metadata (pyproject.toml) ... error
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
Error: module python3-scipy: Child process exited with code 1
```

Do you know how to compile [scipy 1.14.1 from source](https://files.pythonhosted.org/packages/62/11/4d44a1f274e002784e4dbdb81e0ea96d2de2d1045b2132d5af62cc31fd28/scipy-1.14.1.tar.gz) and [numpy 2.1.0](https://files.pythonhosted.org/packages/54/a4/f8188c4f3e07f7737683588210c073478abcb542048cf4ab6fedad0b458a/numpy-2.1.0.tar.gz)?

Reply:
Try setting this for scipy after `buildsystem`:

```json
            "build-options": {
                "env": {
                    "ATLAS": "None",
                    "BLAS": "/app/lib",
                    "LAPACK": "/app/lib"
                }
            },
```

Reply:
what's the issue with numpy ?

Reply:
The numpy version is old and slow 1.26.4.

Reply:
```
		{
		    "name": "python3-scipy",
		    "buildsystem": "simple",
		    "build-options": {
                "env": {
                    "ATLAS": "None",
                    "BLAS": "/app/lib",
                    "LAPACK": "/app/lib"
                }
            },
		    "build-commands": [
		        "pip3 install --verbose --exists-action=i --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} \"scipy\" --no-build-isolation"
		    ],
		    "sources": [
		        {
		            "type": "file",
		            "url": "https://files.pythonhosted.org/packages/30/85/cdbf2c3c460fe5aae812917866392068a88d02f07de0fe31ce738734c477/scipy-1.12.0.tar.gz",
		            "sha256": "4bf5abab8a36d20193c698b0f1fc282c1d083c94723902c447e5d2f1780936a3"
		        }
		    ]
		},
```

```
========================================================================
Building module python3-scipy in /home/<USERNAME>/.var/app/org.gnome.Builder/cache/gnome-builder/flatpak-builder/build/python3-scipy-2
========================================================================
Running: pip3 install --verbose --exists-action=i --no-index --find-links="file://${PWD}" --prefix=${FLATPAK_DEST} "scipy" --no-build-isolation
Using pip 24.1.2 from /usr/lib/python3.11/site-packages/pip (python 3.11)
Looking in links: file:///run/build/python3-scipy
Processing ./scipy-1.12.0.tar.gz
  Running command Preparing metadata (pyproject.toml)
  + meson setup /tmp/pip-install-mdr55mbs/scipy_9a83efb258414406a16a36dbf966e587 /tmp/pip-install-mdr55mbs/scipy_9a83efb258414406a16a36dbf966e587/.mesonpy-4h_b9mmo -Dbuildtype=release -Db_ndebug=if-release -Db_vscrt=md --native-file=/tmp/pip-install-mdr55mbs/scipy_9a83efb258414406a16a36dbf966e587/.mesonpy-4h_b9mmo/meson-python-native-file.ini
  The Meson build system
  Version: 1.3.2
  Source dir: /tmp/pip-install-mdr55mbs/scipy_9a83efb258414406a16a36dbf966e587
  Build dir: /tmp/pip-install-mdr55mbs/scipy_9a83efb258414406a16a36dbf966e587/.mesonpy-4h_b9mmo
  Build type: native build
  Project name: scipy
  Project version: 1.12.0
  C compiler for the host machine: ccache cc (gcc 13.2.0 "cc (GCC) 13.2.0")
  C linker for the host machine: cc ld.bfd 2.42
  C++ compiler for the host machine: ccache c++ (gcc 13.2.0 "c++ (GCC) 13.2.0")
  C++ linker for the host machine: c++ ld.bfd 2.42
  Cython compiler for the host machine: cython (cython 0.29.37)
  Host machine cpu family: x86_64
  Host machine cpu: x86_64
  Program python found: YES (/usr/bin/python)
  Found pkg-config: YES (/usr/bin/pkg-config) 2.2.0
  Run-time dependency python found: YES 3.11
  Program cython found: YES (/usr/bin/cython)
  Compiler for C supports arguments -Wno-unused-but-set-variable: YES
  Compiler for C supports arguments -Wno-unused-function: YES
  Compiler for C supports arguments -Wno-conversion: YES
  Compiler for C supports arguments -Wno-misleading-indentation: YES
  Library m found: YES
  Fortran compiler for the host machine: gfortran (gcc 13.2.0 "GNU Fortran (GCC) 13.2.0")
  Fortran linker for the host machine: gfortran ld.bfd 2.42
  Compiler for Fortran supports arguments -Wno-conversion: YES
  Checking if "-Wl,--version-script" : links: YES
  Program pythran found: YES (/app/bin/pythran)
  Found CMake: /usr/bin/cmake (3.29.6)
  WARNING: CMake Toolchain: Failed to determine CMake compilers state
  Run-time dependency xsimd found: NO (tried pkgconfig and cmake)
  Run-time dependency threads found: YES
  Library npymath found: YES
  Library npyrandom found: YES
  pybind11-config found: YES (/app/bin/pybind11-config) 2.11.1
  Run-time dependency pybind11 found: YES 2.11.1
  Run-time dependency scipy-openblas found: NO (tried pkgconfig)
  Run-time dependency openblas found: NO (tried pkgconfig and cmake)
  Run-time dependency openblas found: NO (tried pkgconfig and cmake)

  ../scipy/meson.build:169:9: ERROR: Dependency "OpenBLAS" not found, tried pkgconfig and cmake

  A full log can be found at /tmp/pip-install-mdr55mbs/scipy_9a83efb258414406a16a36dbf966e587/.mesonpy-4h_b9mmo/meson-logs/meson-log.txt
  error: subprocess-exited-with-error
  
  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /usr/bin/python /usr/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py prepare_metadata_for_build_wheel /tmp/tmpez6wilgr
  cwd: /tmp/pip-install-mdr55mbs/scipy_9a83efb258414406a16a36dbf966e587
  Preparing metadata (pyproject.toml) ... error
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
Error: module python3-scipy: Child process exited with code 1
```

Reply:
can you push the rest of the changes until now? We can figure it out in the PR.

Reply:
Pushing soon.

Reply:
Pushed to one of the other branches. Have a look.

Reply:
Please push it here. I don't know where this other branch is.

Reply:
Added updates to reflect changes.